charmcraft/application/commands/store.py

```diff
@@ -1634,6 +1635,88 @@ def run(self, parsed_args):
             emit.message(cli.format_content(output_data, parsed_args.format))
 
 
+class FetchLibs(CharmcraftCommand):
+    """Fetch libraries defined in charmcraft.yaml."""
+
+    name = "fetch-libs"
+    help_msg = "Fetch one or more charm libraries"
+    overview = textwrap.dedent(
+        """
+        Fetch charm libraries defined in charmcraft.yaml.
+
+        For each library in the top-level `charm-libs` key, fetch the latest library
+        version matching those requirements.
+
+        For example:
+        charm-libs:
+          # Fetch lib with API version 0.
+          # If `fetch-libs` is run and a newer minor version is available,
+          # it will be fetched from the store.
+          - lib: postgresql.postgres_client
+            version: "0"
+          # Always fetch precisely version 0.57.
+          - lib: mysql.client
+            version: "0.57"
+        """
+    )
+    format_option = True
+    always_load_project = True
+
+    def run(self, parsed_args: argparse.Namespace) -> None:
+        """Fetch libraries."""
+        store = self._services.store
+        charm_libs = self._services.project.charm_libs
+        if not charm_libs:
+            raise errors.LibraryError(
+                message="No dependent libraries declared in charmcraft.yaml.",
+                resolution="Add a 'charm-libs' section to charmcraft.yaml.",
+                retcode=78,  # EX_CONFIG: configuration error
+            )
+        emit.progress("Getting library metadata from charmhub")
+        libs_metadata = store.get_libraries_metadata_by_name(charm_libs)
+        declared_libs = {lib.lib: lib for lib in charm_libs}
+        missing_store_libs = declared_libs.keys() - libs_metadata.keys()
+        if missing_store_libs:
+            missing_libs_source = [declared_libs[lib].dict() for lib in missing_store_libs]
+            libs_yaml = util.dump_yaml(missing_libs_source)
+            raise errors.CraftError(
+                f"Could not find the following libraries on charmhub:\n{libs_yaml}",
+                resolution="Use 'charmcraft list-lib' to check library names and versions.",
+                reportable=False,
+                logpath_report=False,
+            )
+
+        emit.trace(f"Library metadata retrieved: {libs_metadata}")
+        local_libs = {
+            f"{lib.charm_name}.{lib.lib_name}": lib for lib in utils.get_libs_from_tree()
+        }
+        emit.trace(f"Local libraries: {local_libs}")
+
+        downloaded_libs = 0
+        for lib_md in libs_metadata.values():
+            local_lib = local_libs.get(f"{lib_md.charm_name}.{lib_md.lib_name}")
+            if local_lib and local_lib.content_hash == lib_md.content_hash:
```

Comment:
Not sure if any stale cache scenarios could occur, but if something like `emit.debug(f"Not downloading {local_lib.lib_name} with hash {lib_md.content_has} because it already exists")` could be useful to devs.

Reply:
Good idea. I've changed the wording and included a suggestion of how to fix it - lmk what you think.

Reply:
Nice!